src/RestApi/StoreApi/Controllers/Cart.php

```diff
@@ -103,8 +103,53 @@ public function register_routes() {
 				'schema' => [ $this, 'get_public_item_schema' ],
 			]
 		);
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/select-shipping',
+			[
+				'args'   => [
+					'context' => $this->get_context_param( [ 'default' => 'view' ] ),
+				],
+				[
+					'methods'  => 'POST',
+					'callback' => [ $this, 'select_shipping' ],
```

Comment:
You can add supported args here too since this expects only the selected rate ids.